pos_pin/static/src/js/pos.js

```diff
@@ -0,0 +1,36 @@
+odoo.define('pos_pin.pos', function (require) {
+    "use strict";
+
+    var screens = require('point_of_sale.screens');
+
+    screens.PaymentScreenWidget.include({
+        check_especial_right: function(option) {
+            /* 
+            Show up a popup with selection users. Check whether array with selected user groups contains especial group.
+            If it does then execute the pin_action function.
+
+            :param dict option: {'title': str, 'error_title': str, 'error_body': str, 'especial_group': int (group_id)}
+            */
+            var self = this;
+            this.gui.select_user({
+                'security': true,
+                'current_user': this.gui.pos.get_cashier(),
+                'title': option.title
+            }).then(function(user) {
+                if ($.inArray(option.especial_group, user.groups_id) >= 0) {
+                    self.pin_action(user);
+                } else {
+                    self.gui.show_popup('error', {
+                        'title': option.error_title,
+                        'body': option.error_body
+                    });
+                }
+            });
+        },
+        pin_action: function(user){
```

Comment:
such approach doesn't allow to use your module in more than one module. Create deffered object instead